### PR TITLE
Fix  bug for guangheng project

### PR DESCRIPTION
**Pull request by triump2020**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #19731 

## What this PR does / why we need it:
1.  Fix bug for reader

### CODE REVIEW

File: pkg/sql/compile/compile.go [hunk 1]
```diff
@@ -4016,6 +4016,9 @@ func (c *Compile) generateNodes(n *plan.Node) (engine.Nodes, []any, []types.T, e
 		if err != nil {
 			return nil, nil, nil, err
 		}
+		if n.TableDef.Name == "defect_history" {
```
Comment (fengttt) [inline pkg/sql/compile/compile.go:4019]: You cannot do this.    This means ALL queries touching this table will generate a log line.   Which  maybe way too excessive.

Comment (fengttt) [reply pkg/sql/compile/compile.go:4019]: And if there are other customer who has a table named defect_history, they will log too.

The right way is always have a session level configuration parameter, if we want to generate this log, we turn it on in a session, run query,  and look at log, then turn it off.

Comment (triump2020) [reply pkg/sql/compile/compile.go:4019]: Yeah ,  i' ll remove all the logs after verify this pr on guangheng product env.

**Comment by triump2020** [comment]
https://github.com/matrixorigin/MO-Cloud/issues/4316

**Review by daviszhen — APPROVED** [review]

**Review by reusee — APPROVED** [review]

**Review by qingxinhome — APPROVED** [review]

**Review by XuPeng-SH — APPROVED** [review]

**Comment by triump2020** [comment]
等待pr 合并

**Comment by triump2020** [comment]
Fix SCA : https://github.com/matrixorigin/linter/pull/32

**Pull request merged**